component/extension.go

```diff
@@ -36,9 +36,7 @@ func UnmarshalExtensionConfig(conf *confmap.Conf, cfg ExtensionConfig) error {
 // Extension is the interface for objects hosted by the OpenTelemetry Collector that
 // don't participate directly on data pipelines but provide some functionality
 // to the service, examples: health check endpoint, z-pages, etc.
-type Extension interface {
-	Component
-}
+type Extension = Component
```

Comment:
What do you think if we do the following?

1. Introduce `Extension` alias to `Component` and add another Extension interface with a temporary name like `TempExtension`

```
type TempExtension interface {
	Component
}

// Deprecated: [0.65.0] Use Component or TempExtension depending on the use case:
// Use Component in Host.GetExtensions() method: GetExtensions() map[component.ID]component.Component
// Use TempExtension in any other case.
type Extension = Component
```

2. Remove the `Extension` alias and move `TempExtension` to `extension.Extension` by deprecating the former.

3. Remove `TempExtension`

This approach may be a bit more disruptive but aligns with our policy for breaking changes.

Reply:
What would we do for `ExtensionFactory.CreateExtension`? Cannot return `TempExtension`.

Reply:
Yes, you're right, I missed that part. It'll still be a breaking change in any way. 

Probably going forward we can notify users about possible issues through a deprecation message for `component.Extension` to make sure that they don't use `component.Component` and `extension.Extension` interchangeably.

Reply:
Will do that.